src/mongocrypt-ctx.c

```diff
@@ -288,21 +287,27 @@ mongocrypt_ctx_setopt_algorithm (mongocrypt_ctx_t *ctx,
                        algorithm);
    }
 
-   if (calculated_len == ALGORITHM_DETERMINISTIC_LEN &&
-       strncmp (algorithm,
-                ALGORITHM_DETERMINISTIC,
-                ALGORITHM_DETERMINISTIC_LEN) == 0) {
+   mstr_view algo_str = mstrv_view_data (algorithm, calculated_len);
+   if (mstr_eq (algo_str, mstrv_lit (ALGORITHM_DETERMINISTIC))) {
       ctx->opts.algorithm = MONGOCRYPT_ENCRYPTION_ALGORITHM_DETERMINISTIC;
-      return true;
-   }
-
-   if (calculated_len == ALGORITHM_RANDOM_LEN &&
-       strncmp (algorithm, ALGORITHM_RANDOM, ALGORITHM_RANDOM_LEN) == 0) {
+   } else if (mstr_eq (algo_str, mstrv_lit (ALGORITHM_RANDOM))) {
       ctx->opts.algorithm = MONGOCRYPT_ENCRYPTION_ALGORITHM_RANDOM;
-      return true;
+   } else if (mstr_eq (algo_str, mstrv_lit ("Indexed"))) {
+      ctx->opts.index_type.value = MONGOCRYPT_INDEX_TYPE_EQUALITY;
+      ctx->opts.index_type.set = true;
+   } else if (mstr_eq (algo_str, mstrv_lit ("Unindexed"))) {
```

Comment:
I think "Indexed" and "Unindexed" should still be defined as constants, and ideally used here and in the related tests. That would be most consistent with how we handle ReadConcern levels, which was one of my original points in [DRIVERS-2352](https://jira.mongodb.org/browse/DRIVERS-2352).